hal/tests/TESTS/pin_names/pinvalidate.py

```diff
@@ -0,0 +1,453 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import os
+import re
+import argparse
+import pathlib
+import json
+from tabulate import tabulate
+
+def find_target(target_name = ''):
+    mbed_os_root = os.path.abspath(os.path.join(pathlib.Path(__file__).parent.absolute(), '..', '..', '..', '..'))
+    targets = dict()
+    
+    try:
+        targets_json_file = open(os.path.join(mbed_os_root, 'targets', 'targets.json'))
+        target_data = json.load(targets_json_file)
+    except:
+        exit("Failed to open targets.json")
+
+    for root, dirs, files in os.walk(mbed_os_root):
+        for f in files:
+            if f == "PinNames.h":
+                path = os.path.join(root, f)
+                name_components = re.findall("TARGET_([a-zA-Z0-9_]*)[\/\\\]", path)
+                name = '_'.join(name_components)
+                if not name:
+                    continue
+                if target_name:
+                    if target_name not in name:
+                        continue
+                # try to find corresponding target in targets.json
+                matches = []
+                for target in target_data:
+                    if "public" in target_data[target]:
+                        if target_data[target]['public'] == False:
+                            continue
+                    if target in name:
+                        matches.append(target)
+                if len(matches) == 0:
+                    #print("No match found in targets.json for " + name)
+                    targets[name] = path
+                elif len(matches) == 1:
+                    targets[matches[0]] = path
+                elif len(matches) > 1:
+                    #print("Multipe matches found " + str(matches) + " in targets.json for " + name)
+                    targets[name] = path
+    return targets
+
+def target_has_arduino_form_factor(target_name):
+    mbed_os_root = os.path.abspath(os.path.join(pathlib.Path(__file__).parent.absolute(), '..', '..', '..', '..'))
+    try:
+        targets_json_file = open(os.path.join(mbed_os_root, 'targets', 'targets.json'))
+        target_data = json.load(targets_json_file)
+    except:
+        exit("Failed to open targets.json")
+    
+    if target_name in target_data:
+        if 'supported_form_factors' in target_data[target_name]:
+            form_factors = target_data[target_name]['supported_form_factors']
+            if 'ARDUINO' in form_factors or 'ARDUINO_UNO' in form_factors:
```

Comment:
I think we should remove `ARDUINO` from this.
We want to drive adoption towards `ARDUINO_UNO` as the official standard and discourage usage of `ARDUINO`.

Reply:
From my side, I think it can be OK like this.
For a ARDUINO target, arduino tests is then executed, but it will fail because ARDUINO_UNO_xxx will be not present.
If you remove ARDUINO here, arduino tests is not executed, then nothing will FAIL

Reply:
Good point Jerome, let's leave it as it's.

Reply:
See comments on proposal here: https://github.com/ARMmbed/mbed-os/pull/14012#issuecomment-748924809

I don't think we can have errors on PRs related to Arduino during the transition; however we should display results to understand progress. This probably requires more discussion.